rule_parser.go

```diff
@@ -189,10 +195,17 @@ func (f *FieldMatcher) unmarshal(field *yaml.Node, values *yaml.Node) error {
 
 	switch values.Kind {
 	case yaml.ScalarNode:
-		f.Values = []string{values.Value}
-
+		f.Values = []interface{}{nil}
+		return values.Decode(&f.Values[0])
 	case yaml.SequenceNode:
 		return values.Decode(&f.Values)
+	case yaml.MappingNode:
+		fallthrough
+	case yaml.DocumentNode:
```

Comment:
Do we ever expect to see a DocumentNode? My understanding was a DocumentNode was the top-level _thing_ and you couldn't encounter one as some lower down field type

Reply:
That's probably true. I don't have an issue dropping it. I was just going through each potential type, but now that you mention it you're most likely right.

Reply:
I'm pretty certain you're right here. I've added a commit to remove that case. :+1: